File: T6_Herencia_Trabajadores/src/model/Autonomo.java
package model;

public class Autonomo extends Trabajador{

    private double sueldo;
    private boolean contratodo;

    public Autonomo() {}

    public Autonomo(String nombre, String apellido, String dni, double sueldo, boolean contratodo) {
        super(nombre, apellido, dni);
        this.sueldo = sueldo;
        this.contratodo = contratodo;
    }

    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println("Salario Anual: "+(sueldo*12));
    }

    public double getSueldo() {
        return sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    public boolean isContratodo() {
        return contratodo;
    }

    public void setContratodo(boolean contratodo) {
        this.contratodo = contratodo;
    }
}

File: DAM2_Pro/PMDM/Tema_1/TrabajoDeHerencia/src/model/Autonomo.kt
package model

class Autonomo(nombre: String, apellido: String, dni: String, salario: Double, var seguro: Boolean, var cuota: Int)
    : Trabajador(nombre,apellido,dni,salario){


    fun pedirDescuento() {
        if (seguro) {
            cuota /= 2
            println("Tu cuota se ha dividido entre 2: $cuota")
        } else {
            println("No tienes seguro")
        }
    }


    override fun calcularSalarioNeto(): Double {
        return salario - (cuota * 12)
    }


    override fun mostrarDatos() {
        super.mostrarDatos()
        println("Seguro: ${seguro}")
    }


}